core/src/main/scala/org/apache/spark/deploy/SparkPipelines.scala

```diff
@@ -62,16 +63,16 @@ object SparkPipelines extends Logging {
         if (opt == "--remote") {
           remote = value
         } else if (opt == "--class") {
-          logInfo("--class argument not supported.")
-          throw SparkUserAppException(SparkExitCode.EXIT_FAILURE)
-        } else if (opt == "--conf" &&
-          value.startsWith("spark.api.mode=") &&
-          value != "spark.api.mode=connect") {
-          logInfo(
-            "--spark.api.mode must be 'connect'. " +
-            "Declarative Pipelines currently only supports Spark Connect."
-          )
+          logError("--class argument not supported.")
           throw SparkUserAppException(SparkExitCode.EXIT_FAILURE)
+        } else if (opt == "--conf" && value.startsWith("spark.api.mode=")) {
+          if (value.toLowerCase(Locale.ROOT) != "spark.api.mode=connect") {
+            logError(
+              "--spark.api.mode must be 'connect'. " +
+                "Declarative Pipelines currently only supports Spark Connect."
+            )
+            throw SparkUserAppException(SparkExitCode.EXIT_FAILURE)
+          }
```

Comment:
Since we always append `--conf spark.api.mode=connect` at the end, here we simply ignore the user-passed value.